packages/web3-utils/src/converters.ts

```diff
@@ -355,28 +355,30 @@ export const toWei = (number: Numbers, unit: EtherUnits): string => {
 };
 
 export const toChecksumAddress = (address: Address): string => {
-	if (typeof address === 'undefined') return '';
-
-	if (!/^(0x)?[0-9a-f]{40}$/i.test(address)) {
+	if (!/^(0x)?[0-9a-f]{40}$/i.test(address) && isHexStrict(address)) {
```

Comment:
Should use [isAddress](https://github.com/ChainSafe/web3.js/blob/4.x/packages/web3-utils/src/validation.ts#L215)

Reply:
We cant use `isAddress` as it expects the address to be checksummed

Reply:
Oh yea, then I would suggest we add a second argument to `isAddress` called `checkChecksum` which is default to `true` to remain consistent with `1.x`. In this case we would call `isAddress(address, false)`

Again, this is just to remain consistent across the repository, so we have one place to check how an address is validated